src/implementations/twig/components/inpage-navigation/inpage-navigation.story.js

```diff
@@ -1,112 +1,130 @@
+      },
+      control: {
+        type: 'text',
+      },
+    };
+  });
+
+  argTypes.blockLeft = {
+    name: 'inject a test content block in the left sidebar',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description:
+      'Inpage navigation recalculates the position of the linked content when something gets added in the dom.',
+    table: {
+      category: 'Test content',
+    },
+    control: {
+      type: 'boolean',
+    },
+  };
+  argTypes.blockMain = {
+    name: 'inject a test content block in the main column',
+    type: { name: 'boolean' },
+    defaultValue: false,
+    description:
+      'Inpage navigation recalculates the position of the linked content when something gets added in the dom.',
+    table: {
+      category: 'Test content',
+    },
+    control: {
+      type: 'boolean',
+    },
+  };
+
+  return argTypes;
 };
-// Add an h2 with a paragraph in the main column.
-const btnIdHandler = () => {
-  const numParagraphs = document.querySelectorAll('.ecl-col-l-9 p').length;
-  const position = Math.floor(Math.random() * Math.floor(numParagraphs));
-  const btnIdElement = document.querySelectorAll('.ecl-col-l-9 p')[position];
-  const demoId = Math.random().toString(36).slice(7);
-  const btnIdTag = document.createElement('h2');
-  btnIdTag.classList.add('ecl-u-type-heading-2');
-  btnIdTag.id = `new-${demoId}`;
-  btnIdTag.innerHTML = `New heading ${demoId}`;
-  const btnIdParagraph = document.createElement('p');
-  btnIdParagraph.classList.add('ecl-u-type-paragraph-m');
-  btnIdParagraph.innerHTML = demoText;
-  btnIdElement.insertAdjacentHTML('afterend', btnIdParagraph.outerHTML);
-  btnIdElement.insertAdjacentHTML('afterend', btnIdTag.outerHTML);
-  // Prevent the story from being reloaded.
+
+// Add generic container in the left or main column.
+const blockHandler = (region, state) => {
+  if (state) {
+    return `<div class="ecl-u-bg-grey-25 ecl-u-mb-m ecl-u-pa-l">
+              <h4 class="ecl-u-type-heading-4">This is an injected block in the ${region} column</h4>
+            </div>`;
+  }
+
   return false;
 };
 
-export default {
-  title: 'Components/Navigation/Inpage navigation',
+// Prepare data for the navigation.
+const prepareData = (data, args) => {
+  data.icon_path = defaultSprite;
+  data.links.forEach((item, i) => {
+    item.label = args[`heading${i + 1}`];
+  });
+
+  return data;
 };
 
-export const Default = () => {
-  // Buttons for the demo.
-  button(btnLeftLabel, btnLeftHandler, tabLabels.cases);
-  button(btnMainLabel, btnMainHandler, tabLabels.cases);
-  button(btnIdLabel, btnIdHandler, tabLabels.cases);
-  button(btnIdRemoveLabel, btnIdRemoveHandler, tabLabels.cases);
+// Prepare Html for the main content.
+const prepareHtmlContent = (data, args) => {
+  let html = '';
+  const lorem = loremIpsum({ count: 25 });
 
-  let pageFillerHtml = '';
-  demoData.links.forEach((content) => {
-    pageFillerHtml += content.item;
+  data.links.forEach((item, i) => {
```

Comment:
There's a difference in style compared to the file and even the concatenation items themselves. i.e no need for concatenation, the string literal will do just fine here too

```js
const prepareHtmlContent = (data) => {
  const lorem = loremIpsum({ count: 25 });
  return data.links.map(({ label }, index) => {
    console.log('index', index);
    return `
    <h2 class="ecl-u-type-heading-2" id="inline-nav-${index}">${label}</h2>
    <p class="ecl-u-type-paragraph-m">${lorem}</p>
    <p class="ecl-u-type-paragraph-m">${lorem}</p>
    `;
  });
};
```

Btw, I'm using the demo data directly